### PR TITLE
Short function replacements

### PR DESCRIPTION
**Pull request by ethouris**

### CODE REVIEW

File: srtcore/api.h [hunk 1]
```diff
@@ -70,9 +70,27 @@ class CUDT;
...
+       m_ListenSocket(0),
+       m_PeerID(0),
+       m_iISN(0),
+       m_pUDT(NULL),
+       m_pQueuedSockets(NULL),
+       m_pAcceptSockets(NULL),
+       m_AcceptCond(),
+       m_AcceptLock(),
+       m_uiBackLog(0),
+       m_iMuxID(-1)
```
Comment (maxsharabayko) [inline srtcore/api.h]: Please use the new formatting:
```
CUDTSocket()
    : m_Status(SRTS_INIT)
    , m_SocketID(0)
```
etc.

File: srtcore/api.h [hunk 2]
```diff
@@ -174,12 +199,13 @@ friend class CRendezvousQueue;
 
       // socket APIs
 
-   int bind(const SRTSOCKET u, const sockaddr_any& name);
-   int bind(const SRTSOCKET u, UDPSOCKET udpsock);
+   int bind(CUDTSocket* u, const sockaddr_any& name);
+   int bind(CUDTSocket* u, UDPSOCKET udpsock);
```
Comment (maxsharabayko) [inline srtcore/api.h]: What is the reason to remove `const`?

File: srtcore/api.h (hunk 1 above)
Comment (ethouris) [reply srtcore/api.h]: Right.

File: srtcore/api.cpp [hunk 3]
```diff
@@ -1520,11 +1520,20 @@ void CUDTUnited::removeSocket(const SRTSOCKET u)
       for (set<SRTSOCKET>::iterator q = i->second->m_pQueuedSockets->begin();
          q != i->second->m_pQueuedSockets->end(); ++ q)
       {
-         m_Sockets[*q]->m_pUDT->m_bBroken = true;
-         m_Sockets[*q]->m_pUDT->close();
-         m_Sockets[*q]->m_tsClosureTimeStamp = srt::sync::steady_clock::now();
-         m_Sockets[*q]->m_Status = SRTS_CLOSED;
-         m_ClosedSockets[*q] = m_Sockets[*q];
+         sockets_t::iterator si = m_Sockets.find(*q);
+         if (si == m_Sockets.end())
```
Comment (ethouris) [inline srtcore/api.cpp]: There's a small bugfix here. Previous version was blindly addressing `m_Sockets[*q]` without checking for existence (so, potentially adding an element). The changed version simply does nothing when the element is not there and reports an IPE. This should rather never happen, but it covers a potential execution path that may lead to a crash.

**Pull request merged**